test/TestNamespaceTracker.hs

```diff
@@ -261,108 +241,100 @@ testSubscribeAsClient =
+    ClientData alice' $ Right $ RequestBundle [UMsgSubscribe helloP] [],
     ClientDisconnect alice',
     -- Should miss this message:
-    ClientData alice [msg helloP Set]
+    ClientData alice $ Left $ UpdateBundle [] [Right $ dum helloP Set]
     ]
 
 testClientDisconnectUnsubs = do
     response <- trackerHelper _disconnectUnsubsBase
     assertEqual "alice': init msgs"
-        (Map.singleton alice' [[msg helloP AssignType]])
+        (Map.singleton alice' $ [UpdateBundle [] [Left $ UMsgAssignType helloP helloP]])
         response
 
 testClientDisconnectUnsubsResubs = do
     response <- trackerHelper $ _disconnectUnsubsBase ++ [
-        ClientData alice' [msg helloP Subscribe],
-        ClientData alice [msg helloP Add]]
+        ClientData alice' $ Right $ RequestBundle [UMsgSubscribe helloP] [],
+        ClientData alice $ Left $ UpdateBundle [] [Right $ dum helloP Add]]
     assertEqual "alice': 2 * init msgs + add msg"
-        (Map.singleton alice' [
-            [msg helloP AssignType],
-            [msg helloP AssignType],
-            [msg helloP Add],
-            [msg helloP Delete]]) -- End of test disconnect
+        (Map.singleton alice' $ [
+            UpdateBundle [] [Left $ UMsgAssignType helloP helloP],
+            UpdateBundle [] [Left $ UMsgAssignType helloP helloP],
+            UpdateBundle [] [Right $ dum helloP Add]])
         response
 
 testOwnerDisconnectDisowns = do
     -- and unregisters clients
     response <- trackerHelper [
-        ClientData alice' [msg [pathq|/fudge|] AssignType], -- Need something to disconnect at end
-        ClientData alice [msg helloP AssignType],
-        ClientData alice' [msg helloP Subscribe],
+        ClientData alice' $ Left $ UpdateBundle [] [Left $ UMsgAssignType [pathq|/fudge|] root], -- Need something to disconnect at end
+        ClientData alice $ Left $ UpdateBundle [] [Left $ UMsgAssignType helloP root],
+        ClientData alice' $ Right $ RequestBundle [UMsgSubscribe helloP] [],
         ClientDisconnect alice,
-        -- No owner => unclaimed => Subscribe disallowed:
-        ClientData alice'' [msg helloP Subscribe],
         -- No subscriber => no AssignType msg:
-        ClientData alice''' [msg helloP AssignType]]
+        ClientData alice''' $ Left $ UpdateBundle [] [Left $ UMsgAssignType helloP root],
+        -- New subscriber:
+        ClientData alice'' $ Right $ RequestBundle [UMsgSubscribe helloP] []]
     assertOnlyKeysInMap [alice', alice''] response
     assertMapValue alice' [
-        [msg helloP AssignType], [msg helloP Delete]] response
-    assertSingleError alice'' helloP ["forbidden"] response
+        UpdateBundle [] [Left $ UMsgAssignType helloP helloP], UpdateBundle [] [Left $ UMsgDelete helloP]] response
+    assertMapValue alice'' [UpdateBundle [] [Left $ UMsgAssignType helloP helloP]] response
   where
-    alice'' = newAwu 44 "alice"
-    alice''' = newAwu 45 "alice"
+    alice'' =  "alice''"
+    alice''' = "alice'''"
```

Comment:
Are you just calling all the users "alice" with n primes? Or are these genuinely all required to be Alice at different addresses? Put another way, the original point of having `alice` and `alice'` was that it was the same person connecting from two addresses.

Reply:
I was blindly porting by this point, but no, since this never sees the user there's no distinction.